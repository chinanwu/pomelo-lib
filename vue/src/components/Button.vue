<template>
  <button
    :id="id"
    :class="['Button', cClass]"
    :ariaLabel="ariaLabel"
    :role="role"
    :ariaHaspopup="ariaHasPopup"
    :ariaRoledescription="ariaRoledescription"
    :disabled="disabled"
    @click="handleClick"
    @keydown="handleKeyDown"
  >
    {{ label }}
  </button>
</template>

<script>
export default {
  name: "Button",
  props: {
    id: String,
    label: String,
    cClass: String,
    ariaLabel: String,
    role: String,
    ariaHasPopup: String,
    ariaRoledescription: String,
    disabled: Boolean
  },
  emits: ["click", "keydown"],
  methods: {
    handleClick() {
      this.$emit("click");
    },
    handleKeyDown() {
      this.$emit("keydown");
    }
  }
};
</script>

<style lang="less" scoped>
.Button {
  background-color: inherit;
  border: none;
  cursor: pointer;
}
</style>
