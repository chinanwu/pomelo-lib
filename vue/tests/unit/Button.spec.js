import { shallowMount } from "@vue/test-utils";
import Button from "@/components/Button.vue";

describe("Button.vue", () => {
  it("renders Button", () => {
    const wrapper = shallowMount(Button);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders Button with props", () => {
    const wrapper = shallowMount(Button, {
      props: {
        id: "test",
        class: "Button--fake",
        ariaLabel: "Fake Button",
        role: "link",
        ariaHasPopup: "dialog",
        ariaRoledescription: "this is unnecessary but test",
        disabled: false
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("emits click", () => {
    const wrapper = shallowMount(Button, {
      props: {
        id: "test"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.find("#test").trigger("click");
    expect(wrapper.emitted()).toHaveProperty("click");
  });

  it("emits keydown", () => {
    const wrapper = shallowMount(Button, {
      props: {
        id: "test"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.find("#test").trigger("keydown");
    expect(wrapper.emitted()).toHaveProperty("keydown");
  });
});
