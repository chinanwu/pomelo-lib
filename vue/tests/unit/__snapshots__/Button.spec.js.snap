// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue emits click 1`] = `
<button
  class="Button"
  id="test"
/>
`;

exports[`Button.vue emits keydown 1`] = `
<button
  class="Button"
  id="test"
/>
`;

exports[`Button.vue renders Button 1`] = `
<button
  class="Button"
/>
`;

exports[`Button.vue renders Button with props 1`] = `
<button
  ariahaspopup="dialog"
  arialabel="Fake Button"
  ariaroledescription="this is unnecessary but test"
  class="Button Button--fake"
  id="test"
  role="link"
/>
`;
